### PR TITLE
Collection content item filters

### PR DESCRIPTION
**Pull request by kdoh**
Adds back item content filters if there are more than 5 unpinned items and more than one type of item (e.g. dashboards and questions). I think eventually we'll want to rework this interaction but for now our old tab bar helps narrow things down in large collections in a way that we were missing in the new layout.

image

### CODE REVIEW

File: frontend/src/metabase/collections/containers/CollectionContent.jsx [hunk 1]
```diff
@@ -284,9 +290,30 @@ export default class CollectionContent extends React.Component {
             </PinDropTarget>
           )}
           <Box className="relative" mt={1}>
-            {// if there are pins and we also have items show a heading for this section
-            collectionHasPins && unpinnedItems.length > 0 && (
-              <CollectionSectionHeading>{t`Everything else`}</CollectionSectionHeading>
+            {showFilters ? (
+              <ItemTypeFilterBar
+                analyticsContext={ANALYTICS_CONTEXT}
+                filters={ITEM_TYPE_FILTERS.map((f, i) => {
```
Comment (paulrosenzweig) [inline frontend/src/metabase/collections/containers/CollectionContent.jsx:296]: Maybe `assocIn` is more concise/readable?
```js
const everythingName= collectionHasPins && unpinned.length > 0 ? t`Everything else` : t`Everything`;
const filters = assocIn(ITEM_TYPE_FILTERS, [0, 'name'], everythingName)
```

**Review by paulrosenzweig — APPROVED** [review]
Looks good!

File: frontend/src/metabase/collections/containers/CollectionContent.jsx (hunk 1 above)
Comment (kdoh) [reply frontend/src/metabase/collections/containers/CollectionContent.jsx]: Yeah it sure is!

**Comment by mazameli** [comment]
Dunno that we need the stroke above the tabs in this state:

image

**Comment by mazameli** [comment]
Super minor, but there's more space above the first item here than below it; easier to see when hovering:

image

**Pull request merged**